mvrx/src/main/kotlin/com/airbnb/mvrx/DeliveryMode.kt

```diff
@@ -0,0 +1,39 @@
+package com.airbnb.mvrx
+
+import androidx.annotation.RestrictTo
+import kotlin.reflect.KProperty1
+
+/**
+ * Defines what updates a subscription should receive.
+ * See: [RedeliverOnStart], [UniqueOnly].
+ */
+sealed class DeliveryMode {
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    fun appendPropertiesToId(vararg properties: KProperty1<*, *>): DeliveryMode {
+        return when (this) {
+            is RedeliverOnStart -> RedeliverOnStart
+            is UniqueOnly -> UniqueOnly(subscriptionId + "_" + properties.joinToString(",") { it.name })
```

Comment:
nit: 
```suggestion
            is UniqueOnly -> UniqueOnly(properties.joinToString(",", prefix = subscriptionId + "_") { it.name })
```
will allocate slightly less memory